src/libraries/System.Net.Quic/src/System/Net/Quic/Internal/MsQuicTlsSecret.cs

```diff
@@ -0,0 +1,106 @@
+
+                    if (status == QUIC_STATUS_SUCCESS)
+                    {
+                        return new MsQuicTlsSecret(ptr);
+                    }
+                    else
+                    {
+                        NetEventSource.Error(handle, "Failed to set native memory for TLS secret.");
+                    }
+                }
+            }
+            catch { };
+        }
+
+        return null;
+    }
+
+    private unsafe MsQuicTlsSecret(QUIC_TLS_SECRETS* memory)
+    {
+        _tlsSecrets = memory;
+    }
+
+    public void WriteSecret() => WriteSecret(s_fileStream);
+    public unsafe void WriteSecret(FileStream? stream)
+    {
+        if (stream != null && _tlsSecrets != null)
+        {
+            lock (stream)
+            {
+                string clientRandom = string.Empty;
+
+                if (_tlsSecrets->IsSet.ClientRandom != 0)
+                {
+                    clientRandom = HexConverter.ToString(new ReadOnlySpan<byte>(_tlsSecrets->ClientRandom, 32));
+                }
+
+                if (_tlsSecrets->IsSet.ClientHandshakeTrafficSecret != 0)
+                {
+                    stream.Write(Encoding.ASCII.GetBytes($"CLIENT_HANDSHAKE_TRAFFIC_SECRET {clientRandom} {HexConverter.ToString(new ReadOnlySpan<byte>(_tlsSecrets->ClientHandshakeTrafficSecret, _tlsSecrets->SecretLength))}\n"));
+                }
+
+                if (_tlsSecrets->IsSet.ServerHandshakeTrafficSecret != 0)
+                {
+                    stream.Write(Encoding.ASCII.GetBytes($"SERVER_HANDSHAKE_TRAFFIC_SECRET {clientRandom} {HexConverter.ToString(new ReadOnlySpan<byte>(_tlsSecrets->ServerHandshakeTrafficSecret, _tlsSecrets->SecretLength))}\n"));
+                }
+
+                if (_tlsSecrets->IsSet.ClientTrafficSecret0 != 0)
+                {
+                    stream.Write(Encoding.ASCII.GetBytes($"CLIENT_TRAFFIC_SECRET_0 {clientRandom} {HexConverter.ToString(new ReadOnlySpan<byte>(_tlsSecrets->ClientTrafficSecret0, _tlsSecrets->SecretLength))}\n"));
+                }
+
+                if (_tlsSecrets->IsSet.ServerTrafficSecret0 != 0)
+                {
+                    stream.Write(Encoding.ASCII.GetBytes($"SERVER_TRAFFIC_SECRET_0 {clientRandom} {HexConverter.ToString(new ReadOnlySpan<byte>(_tlsSecrets->ServerTrafficSecret0, _tlsSecrets->SecretLength))}\n"));
+                }
+
+                if (_tlsSecrets->IsSet.ClientEarlyTrafficSecret != 0)
+                {
+                    stream.Write(Encoding.ASCII.GetBytes($"CLIENT_EARLY_TRAFFIC_SECRET {clientRandom} {HexConverter.ToString(new ReadOnlySpan<byte>(_tlsSecrets->ClientEarlyTrafficSecret, _tlsSecrets->SecretLength))}\n"));
+                }
+
+                stream.Flush();
+            }
+        }
+    }
+
+    public unsafe void Dispose()
+    {
+        if (_tlsSecrets != null)
+        {
+            NativeMemory.Clear(_tlsSecrets, (nuint)sizeof(QUIC_TLS_SECRETS));
```

Comment:
What is the value of clearing without freeing memory?

Reply:
the goal was to minimize exposure of the secrets. After the `Clear` the actual secrets should disappear from core dump.